### PR TITLE
mzconduct: merge into mzcompose

### PR DESCRIPTION
**Pull request by benesch**
Merge mzconduct and mzcompose. Where you previously ran

    $ bin/mzconduct chbench --workflow=ci

you now run:

    $ cd demo/chbench && ./mzcompose run ci

The hope is that the new API is less confusing to users. Everything
happens locally in the demo/test directory, so you no longer need to
reach out to ../../bin/mzconduct for some commands and ./mzcompose for
others.

The code is also greatly simplified, since we no longer have two
separate programs that need to parse and understand mzcompose.yml files.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5072)

### CODE REVIEW

**Comment by benesch** [comment]
This diff goes down much easier if you hide whitespace changes!

File: misc/python/materialize/cli/mzcompose.py [hunk 1]
```diff
@@ -31,117 +29,99 @@
...
+        for name in repo.compositions:
+            print(name)
+        return 0
+
+    # Load composition.
+    try:
+        composition = mzcompose.Composition(repo, args.mz_find or Path.cwd().name)
+    except errors.UnknownComposition:
+        if args.mz_find:
+            print(f'unknown composition "{args.mz_find}"', file=sys.stderr)
```
Comment (quodlibetor) [inline misc/python/materialize/cli/mzcompose.py:59]: TIOLI: it's slightly more robust to force a `repr` format with `!r`:

```suggestion
            print(f'unknown composition {args.mz_find!r}', file=sys.stderr)
```

```python
>>> v = 'one"\'two'
>>> print(f"{v!r}")
'one"\'two'
>>> c = 'three four'
>>> print(f"{c!r}")
'three four'
```

File: misc/python/materialize/mzcompose.py [hunk 2]
```diff
@@ -66,291 +64,139 @@
...
-    """Show the directory that this composition is in"""
-    print(str(Composition.find(composition).path))
-
-
-# Composition Discovery
-
-
 class Composition:
-    """Information about an mzcompose instance
+    """A set of parsed mzcompose.yml files."""
```
Comment (quodlibetor) [inline misc/python/materialize/mzcompose.py:68]: This is the set of mzcompose.yml files for a single directory, right?

**Review by quodlibetor — APPROVED** [review]
LGTM!

File: misc/python/materialize/mzcompose.py (hunk 2 above)
Comment (benesch) [reply misc/python/materialize/mzcompose.py:68]: Good point, yeah.

File: misc/python/materialize/cli/mzcompose.py (hunk 1 above)
Comment (benesch) [reply misc/python/materialize/cli/mzcompose.py:59]: ah, done.

File: demo/chbench/bin/generate_snapshot [hunk 3]
```diff
@@ -10,11 +10,6 @@
 
 set -euo pipefail
```
[the remark this answers is not in the document]
Comment (benesch) [reply demo/chbench/bin/generate_snapshot:12]: Done.

**Pull request merged**